spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteDataFilesSparkAction.java

```diff
@@ -168,89 +160,36 @@ public RewriteDataFiles.Result execute() {
 
     validateAndInitOptions();
 
-    StructLikeMap<List<List<FileScanTask>>> fileGroupsByPartition =
-        planFileGroups(startingSnapshotId);
-    RewriteExecutionContext ctx = new RewriteExecutionContext(fileGroupsByPartition);
+    RewritePlanResult result = plan(startingSnapshotId);
```

Comment:
minor nit here because we have a "result" variable which is not actually the "result" of this function but is actually just an intermediary input. We could probably change the class name to just RewritePlan and the variable to just plan